src/mbgl/map/source.cpp

```diff
@@ -431,7 +460,10 @@ void Source::update(MapData& data,
         bool obsolete = std::find(retain.begin(), retain.end(), tile.id) == retain.end();
         if (!obsolete) {
             retain_data.insert(tile.data->id);
-        } else if (type != SourceType::Raster && tile.data->ready()) {
+        } else if (type != SourceType::Raster && tile.data->state == TileData::State::parsed) {
```

Comment:
Along the lines of `TileData::isReadyState()`, can we move this into a function too?